optimum/habana/transformers/generation/utils.py

```diff
@@ -2371,7 +2380,6 @@ def constrained_beam_search(
                 cur_len = cur_len + 1
                 continue  # don't waste resources running the code we don't need
 
-            token_idx = model_kwargs.get("token_idx", None)
```

Comment:
Why removing this line? We need to update `token_idx` at each iteration if I'm not mistaken.

Reply:
We get it once out side the loop and set cur_len to it. 
cur_len is incremented inside the loop and use cur_len to slice the tensor.
Note that the logic will now work even if we don't set static shapes as earlier code was forcefully using token_idx.item() inside the loop..

<<
 token_idx = model_kwargs.get("token_idx", None)
 if token_idx is not None:
     # Update cur_len in case of static shapes
     cur_len = token_idx.item()
>>

Reply:
Logic is now same in both constrained_beam_search() as well as beam_search()

Reply:
token_idx is still updated though by calling update_model_kwargs(). It is just we don't need to read it here since we can use cur_len.

Reply:
Ah yes you're fully right, I missed that `token_idx` is updated anyway. Thanks for the explanation!